### PR TITLE
[DownloadBuildArtifactsV0] Replace decompress-zip with extract-zip package

### PR DESCRIPTION
**Pull request by kirill-ivlev**
**Task name**: DownloadBuildArtifactsV0

**Description**: Replace decompress-zip package with extract-zip package to avoid incapability issues with Node17+

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) https://mseng.visualstudio.com/AzureDevOps/_workitems/edit/2150886 
https://mseng.visualstudio.com/AzureDevOps/_workitems/edit/2150886
**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by DmitriiBobreshev — APPROVED** [review]

**Review by ivanduplenskikh — APPROVED** [review]

**Review by DergachevE — APPROVED** [review]

File: _generated/DownloadBuildArtifactsV0_Node20/package.json [hunk 1]
```diff
@@ -22,7 +22,8 @@
     "artifact-engine": "^1.4.0",
     "azure-devops-node-api": "^10.2.2",
     "azure-pipelines-task-lib": "^4.4.0",
-    "decompress-zip": "0.3.3"
+    "decompress-zip": "0.3.3",
```
Comment (KonstantinTyukalov) [inline _generated/DownloadBuildArtifactsV0_Node20/package.json:25]: Are we need decompress-zip now?

**Review by manolerazvan — APPROVED** [review]

File: Tasks/DownloadBuildArtifactsV0/_buildConfigs/Node20/package.json [hunk 2]
```diff
@@ -22,7 +22,8 @@
     "artifact-engine": "^1.4.0",
     "azure-devops-node-api": "^10.2.2",
     "azure-pipelines-task-lib": "^4.4.0",
-    "decompress-zip": "0.3.3"
+    "decompress-zip": "0.3.3",
```
Comment (merlynomsft) [inline Tasks/DownloadBuildArtifactsV0/_buildConfigs/Node20/package.json:25]: We'll need to remember to remove decompress-zip later

File: _generated/DownloadBuildArtifactsV0_Node20/package.json (hunk 1 above)
Comment (kirill-ivlev) [reply _generated/DownloadBuildArtifactsV0_Node20/package.json:25]: will be removed after testing with FF

File: Tasks/DownloadBuildArtifactsV0/DownloadHandlers/DownloadHandlerContainerZip.ts [hunk 3]
```diff
@@ -39,9 +42,15 @@ export class DownloadHandlerContainerZip extends DownloadHandler {
             }
 
             tl.debug(`Extracting ${this.zipLocation} to ${unzipLocation}`);
-
+#if NODE20
+            tl.debug(`Using extract-zip package for extracting archive`);
+            extract(this.zipLocation, { dir: unzipLocation }).then(() => {
+                resolve();
```
Comment (merlynomsft) [inline Tasks/DownloadBuildArtifactsV0/DownloadHandlers/DownloadHandlerContainerZip.ts:48]: should this be return resolve() and return reject(...)?

**Pull request merged**